### ISSUE TITLE
RFE: logind should fall back to creating seat matching rules in /run/ if /etc/ is read-only

### ISSUE THREAD

**Issue by akamaus** [systemd/systemd#15781]
**Is your feature request related to a problem? Please describe.**
Some distributives (eg. NixOS) keep /etc read-only. Currently It makes very hard to setup a multiseat systeam, because loginctl fails with `"Could not attach device: Read-only file system"` error.
Here is a thread about it. https://github.com/NixOS/nixpkgs/issues/86581

**Describe the solution you'd like**
What about gracefully writing rules to `/var/run/udev/rules.d` in case writing to `/etc/udev/rules.d` fails?
Warning user about ephemeral nature of changes would be nice too.

**Describe alternatives you've considered**
Alternative way of setting up multiseat is to do it old way. You ignore seats, you write custom X config, there you disable device auto-detection and manually assign devices to several ServerLayouts. It works, but is very fragile.

**Comment by poettering** [comment]
/var is not an option really. hw an show up any time, including (in fact: typically) during early boot, and /var is not necessarily available then.

/var/run/ in particular is just a symlink to /run, and that is volatile, hence what you are asking for means you'd lose all data when rebooting. 

Configuration belongs in /etc/, and I think seat definitions are configuration, hence I don't think the current behaviour is much of a problem. I mean, there must be some way to make config changes in nixos, for example chaning the hostname or such?

**Comment by akamaus** [comment]
We are speaking about config changes occurring because of user issued `loginctl attach`, so it's not a case of early boot.

I totally agree what seat definitions are part of configuration. But it feels not as fixed as, say, hostname or disk partitions. NixOS uses rather extreme approach. All configuration is written as a program using a special language. After that it's compiled into a bunch of classic config files in a read-only partition, which are linked to /etc. So we have rollback functionality and other goodies.

Unfortunately there is a kind of conflict between NixOS way of doing things and using loginctl for tweaking seats. As a NixOS user I would be happy to create ephemeral configuration (so I'm sure I can just reboot in case I messed things up and monitor turned black because I attached wrong gpu), test it and then manually copy it to nix config. Writing it manually without loginctl is problematic because all the manuals related to multiseat in Linux instruct to use loginctl. And precise format of udev rules is usually not described.

So I kindly ask to consider implementing fall back write to `/var/run/udev` if `/etc/` is found to be mounted read-only. A warning could be shown to users in that case, so they won't be surprised configuration won't survive a reboot.

**Comment by poettering** [comment]
seat assignments are configuration. You put together a seat today, and then reboot, and it still applies, because this works by writing udev rules. And on that subsequent boot the rules are applied during early boot.

**Comment by poettering** [comment]
Sorry, /var is not an option. And /var/run/ is  a legacy alias for /run and volatile.

**Comment by poettering** [comment]
a patch that makes the code in logind fall back to writing a /run udev rule if /etc cannot be written to is something I'd be happy to review and merge though.

But of course, such a fallback would mean any seat definitions are lost on reboot.

**Comment by flokli** [comment]
Yeah, I proposed using `/run/udev/rules.d` initially in https://github.com/NixOS/nixpkgs/issues/86581#issuecomment-623425405, in a NixOS context. I don't think `/var` was really meant here.

Instead of silently writing to `/run/udev/rules.d`, discussed patch could show the filename that the rule was written to to the user, and warn these things would be lost on reboot if not persisted - still all distro-agnostic. 

NixOS users could then just copy over that rule to their `services.udev.extraRules` to persist it to `/etc`.

**Comment by mrhappy200** [comment]
I am still facing this issue on Nixos and was wondering if there was any progress on this issue?
